packages/node_modules/@ciscospark/plugin-messages/src/messages.js

```diff
@@ -28,6 +41,72 @@ import {isArray} from 'lodash';
  * @class
  */
 const Messages = SparkPlugin.extend({
+  /**
+   * Initializer
+   * @param  {...any} args
+   * @returns {undefined}
+   */
+  initialize(...args) {
+    Reflect.apply(SparkPlugin.prototype.initialize, this, args);
+
+    // Listen to new activities from Mercury.
+    this.listenTo(this.spark.internal.mercury, 'event:conversation.activity',
+      (event) => this.onConversationActivityEvent(event));
+  },
+
+  /**
+   * Connect to the web socket to listen to incoming messages.
+   * @returns {Promise}
+   */
+  listen() {
+    return this.spark.internal.mercury.connect();
+  },
+
+  /**
+   * Trigger a "messages:created" event.
+   * @param {Object} event
+   * @returns {undefined}
+   */
+  onConversationActivityEvent(event) {
+    const {activity} = event.data;
+
+    // Reply activities are not currently supported
+    if (activity.type === API_ACTIVITY_TYPE.REPLY) {
```

Comment:
Are these threaded replies?

Reply:
Yes. I figured I'd handle that while I was in there.